services/wallet/async/async_test.go

```diff
@@ -0,0 +1,59 @@
+package async
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAtomicGroupTerminatesOnOneCommandFailed(t *testing.T) {
+	ctx := context.Background()
+	group := NewAtomicGroup(ctx)
+
+	err := errors.New("error")
+	group.Add(func(ctx context.Context) error {
+		time.Sleep(1 * time.Millisecond)
+		return err // failure
+	})
+	group.Add(func(ctx context.Context) error {
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-time.After(1 * time.Millisecond): // wait for the first command to fail and terminate the second command
+				_ = 1
+			}
+		}
+	})
+
+	<-group.WaitAsync()
```

Comment:
What do you think of calling `group.Wait()`? should be enough to prove the point?